assets/js/analyzer.js

```diff
@@ -157,81 +157,78 @@ function analyze() {
     if(currentAnalyzerRequest) {
         currentAnalyzerRequest.abort();
     }
-    currentAnalyzerRequest = $.jsonp({
-        url: config.APY_URL + '/analyze',
-        pageCache: true,
-        beforeSend: ajaxSend,
-        complete: function () {
-            ajaxComplete();
-            currentAnalyzerRequest = undefined;
-        },
-        data: {
-            'lang': analyzerMode,
-            'q': $('#morphAnalyzerInput').val()
-        },
-        success: function (data) {
-            var regex = /([^<]*)((<[^>]+>)*)/g;
-            $('#morphAnalyzerOutput').empty();
-            for(var i = 0; i < data.length; i++) {
-                var leftTD = $('<td class="text-right"></td>');
-                var strong = $('<strong></strong>').text(data[i][1].trim());
-                var arrow = $('<span></span>').html('&nbsp;&nbsp;&#8620;');
-                leftTD.append(strong).append(arrow);
-
-                var rightTD = $('<td class="text-left"></td>');
-                var splitUnit = data[i][0].split('/');
-
-                if(splitUnit[1][0] === '*') {
-                    rightTD.addClass('text-danger');
-                }
 
-                var tr = $('<tr></tr>').append(leftTD).append(rightTD);
-                $('#morphAnalyzerOutput').append(tr);
-
-                var joinedMorphemes = {};
-                for(var j = 1; j < splitUnit.length; j++) {
-                    var unit = splitUnit[j];
-                    if(unit.match(regex).length > 2) {
-                        var matches = unit.match(regex);
-                        for(var k = 1; k < matches.length - 1; k++) {
-                            if(joinedMorphemes[matches[k]]) {
-                                joinedMorphemes[matches[k]].push(unit);
-                            }
-                            else {
-                                joinedMorphemes[matches[k]] = [unit];
-                            }
-                        }
+    var request = {'lang': analyzerMode};
+    request.dataInput = $('#morphAnalyzerInput').val();
+    var methodType = (request.dataInput.length > THRESHOLD_REQUEST_LENGTH) ? 'POST' : 'GET';
```

Comment:
Um, when I create a request object for generator and analyzer modes, I make use of a JSON structure, and name my data variable as `q` ( similar to translator.js request object)! However, Hound throws an error of variable Identifier being too small ( atleast the length must be 4). Therefore, I renamed it to something like `dataInput`. I planned to make the check internal to the generalised function only ( as it is repetitive) but this is causing limitation. How do I resolve this?